tests/cli-paths.rs

```diff
@@ -0,0 +1,403 @@
+//! This file contains tests relevant to Rustup's handling of updating PATHs.
+//! It depends on self-update working, so if absolutely everything here breaks,
+//! check those tests as well.
+pub mod mock;
+
+use crate::mock::clitools::{self, expect_ok, Config, Scenario};
+use crate::mock::{get_path, restore_path};
+use lazy_static::lazy_static;
+use std::path::PathBuf;
+use std::sync::Mutex;
+
+pub fn setup(f: &dyn Fn(&Config)) {
+    clitools::setup(Scenario::SimpleV2, &|config| {
+        // Lock protects environment variables
+        lazy_static! {
+            static ref LOCK: Mutex<()> = Mutex::new(());
+        }
+        let _g = LOCK.lock();
```

Comment:
This lock is only relevant for PATH modifying tests on windows; I suggest not calling setup() from the unix tests